src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -55,6 +56,9 @@ export default class DeployFromSettingsController {
     /** @export {string} */
     this.containerImage = '';
 
+    /** @export {!backendApi.SecretSpec} */
+    this.imagePullSecret = {name: '', namespace: '', type: '', data: {}};
```

Comment:
I think this should be nullable and you should initialize with null to indicate empty-ness. Dont use empty object.